frontend/src/scenes/pipeline/hogfunctions/list/HogFunctionTemplateList.tsx

```diff
@@ -37,71 +44,74 @@ export function HogFunctionTemplateList({
                 {extraControls}
             </div>
 
-            <LemonTable
-                dataSource={filteredTemplates}
-                size="small"
-                loading={loading}
-                columns={[
-                    {
-                        title: '',
-                        width: 0,
-                        render: function RenderIcon(_, template) {
-                            return <HogFunctionIcon src={template.icon_url} size="small" />
+            <OptionalPayGateMini feature={feature} currentUsage={currentUsage} className="mb-2">
```

Comment:
Maybe i've missed the point but why is there a limit on number of alerts?

Reply:
Just for context because I hadn't originally added it to the description. When a user has reached the free tier limit they will be showing the pay gate when trying to create a new notification:
![limit](https://github.com/user-attachments/assets/d960204e-21bc-432c-8db2-c59aa8f7df8f)

In cases where they haven't reached the limit or the `LinkedHogFunctions` doesn't have a feature specified it just renders as normal:
![before](https://github.com/user-attachments/assets/b4c15193-aba8-4a06-906e-ce3a66a794bf)

Added this inside the component because 1) the count of current usage is easier to calculate (given it already fetches the hog functions) and 2) the upgrade UI isn't visible until you actively click "New notification"